core/src/Cache/CacheFallbackOperations.cs

```diff
@@ -172,28 +172,46 @@ internal class CacheFallbackOperations
             return Tuple.Create(clientInfoToAdalUserMap, adalUsersWithoutClientInfo);
         }
 
+        /// <summary>
+        /// Algorithm to delete: 
+        /// 
+        /// Invariant: displayableId is not null 
+        /// 
+        /// if identifier != null then delete everything with the same client info and env
+        /// otherwise, delete everything with the same displayableId
+        /// 
+        /// Notes: 
+        /// - displayableId can change rarely
+        /// - ClientCredential Grant uses the app token cache, not the user token cache, so this algorithm does not apply
+        /// (nor will GetAccounts / RemoveAccount work)
+        /// 
+        /// </summary>
+        /// <param name="legacyCachePersistance"></param>
+        /// <param name="displayableId"></param>
+        /// <param name="environmentAliases"></param>
+        /// <param name="identifier"></param>
         public static void RemoveAdalUser(ILegacyCachePersistance legacyCachePersistance,
             string displayableId, ISet<string> environmentAliases, string identifier)
         {
+            if (string.IsNullOrEmpty(displayableId))
+            {
+                throw CoreExceptionFactory.Instance.GetClientException(
+                    CoreErrorCodes.InternalError,
+                    "Internal error - trying to remove an ADAL user with an empty username. Please log a bug.");
```

Comment:
> Please log a bug. [](start = 91, length = 18)

please remove this

Reply:
I don't agree with this in general (there are just so many use cases in ADAL / MSAL cache removal that it is possible we missed smth and it's safer to fail fast), but I think in this circumstance it is ok (because it is possible our current msal 2.0.0-preview users already have accounts with null displayableId)

Reply:
You removed too much. Using CF would have showed you exactly that I only requested 4 words to be removed here.

---
In reply to: [217615404](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1256#discussion_r217615404) [](ancestors = 217615404)

Reply:
Pls. see my clarification. This was not what I meant.

---
In reply to: [217763505](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1256#discussion_r217763505) [](ancestors = 217763505)